storage/client/queries/queries.go

```diff
@@ -56,10 +56,16 @@ const (
 			($2::bigint IS NULL OR height <= $2::bigint) AND
 			($3::timestamptz IS NULL OR time >= $3::timestamptz) AND
 			($4::timestamptz IS NULL OR time < $4::timestamptz) AND
-			($5::text IS NULL OR block_hash = $5::text)
+			($5::text IS NULL OR block_hash = $5::text) AND
+			($6::text IS NULL OR proposer_entity_id = (
+				SELECT id
+				FROM chain.entities
+				WHERE address = $6::text
+				LIMIT 1
```

Comment:
Nice, looks much sleeker :) One small thing - could you add a `COALESCE( ..., 'none')` or something similar to avoid returning blocks with a `null` block proposer for non-entity addresses?

Reply:
Good idea, updated!